shell/platform/windows/win32_window.cc

```diff
@@ -4,9 +4,18 @@
 
 #include "flutter/shell/platform/windows/win32_window.h"
 
+#include "dpi_utils.h"
+
 namespace flutter {
 
-Win32Window::Win32Window() {}
+Win32Window::Win32Window() {
+  // Get the DPI of the primary monitor as the initial DPI. If Per-Monitor V2 is
+  // supported, |current_dpi_| should be updated in the
+  // kWmDpiChangedBeforeParent message.
```

Comment:
Is `kWmDpiChangedBeforeParent` always called once then? (I.e., if a window is created on a non-primary monitor, the right thing will still happen?)

Reply:
Yeah, if the primary monitor's DPI is different from the one the app is running on, DPI_CHANGED will be called. This is only on V2 since `kWmDpiChangedBeforeParent ` is not called in V1.